File: src/Narfu/Models/Lesson.cs
using System;

namespace Narfu.Models
{
    public class Lesson : IEquatable<Lesson>
    {
        public string Type { get; set; }
        public string Name { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public string StartEndTime { get; set; }
        public byte Number { get; set; }
        public string Address { get; set; }
        public string Auditory { get; set; }
        public string Teacher { get; set; }
        public string Groups { get; set; }

        public bool Equals(Lesson other)
        {
            return StartTime == other.StartTime && Name == other.Name;
        }

        public override int GetHashCode()
        {
            return Type.GetHashCode() ^ StartTime.GetHashCode();
        }
    }
}

File: src/frontend/src/app/models/lesson.ts
import { LessonType } from "./lesson-type";

export interface Lesson {
    id: string;
    type: string;
    name: string;
    startTime: Date;
    endTime: Date;
    startEndTime: string;
    number: number;
    address: string;
    auditory: string;
    teacher: string;
    groups: string;
    lessonType: LessonType;

    link: string;
}